### PR TITLE
Prepare for rust 1.62.1

### PR DESCRIPTION
**Pull request by skunert**
- update UI tests for rust 1.62.1

### CODE REVIEW

**Review by KiChjang — APPROVED** [review]

**Review by sergejparity — APPROVED** [review]

**Comment by alvicsam** [comment]
clippy is not happy with the latest nightly https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/1695838

**Comment by skunert** [comment]
I had to replace some automatic derives of `scale_info::TypeInfo`. I introduced manual implementations for the two warnings we saw (implementation is `cargo-expand` output mostly) and fixed the warnings.

Also clippy should be fixed.

File: primitives/runtime/src/bounded/bounded_vec.rs [hunk 1]
```diff
@@ -104,10 +104,45 @@ where
...
 /// Similar to a `BoundedVec`, but not owned and cannot be decoded.
-#[derive(Encode, scale_info::TypeInfo)]
-#[scale_info(skip_type_params(S))]
+#[derive(Encode)]
 pub struct BoundedSlice<'a, T, S>(pub(super) &'a [T], PhantomData<S>);
 
+// This can be replaced with
+// #[derive(scale_info::TypeInfo)]
+// #[scale_info(skip_type_params(S))]
+// again once this issue is fixed in the rust compiler: https://github.com/rust-lang/rust/issues/96956
```
Comment (bkchr) [inline primitives/runtime/src/bounded/bounded_vec.rs:113]: Please create an issue for this.

Comment (bkchr) [reply primitives/runtime/src/bounded/bounded_vec.rs:113]: So, that we don't forget it.

**Review by bkchr — APPROVED** [review]

File: primitives/runtime/src/bounded/bounded_vec.rs (hunk 1 above)
Comment (skunert) [reply primitives/runtime/src/bounded/bounded_vec.rs:113]: https://github.com/paritytech/substrate/issues/11915, I will also put it in the comment here

**Review by rcny — APPROVED** [review]

**Review by radupopa2010 — APPROVED** [review]

**Pull request merged**